### PR TITLE
feat: add provider-specific ignore file support

### LINKED ISSUE #142: Feature: deploy agent-specific ignore files (.clineignore, .gooseignore)
Deploy agent-specific ignore files controlling which files are excluded from agent context.

**Affected providers (2):** Cline (`.clineignore`), Goose (`.gooseignore`)

Simple content-only feature — no frontmatter, no parsing needed. Could be implemented as a trivial `IgnoreFeature` or handled as a plain-text instruction variant.

### PR DESCRIPTION
**Pull request by soorya-u**
## Summary

- Adds `ignore` feature type to deploy pipeline alongside commands, instructions, MCP, and skills
- Introduces `IgnoreFeature` implementing `FeatureTrait` — loads patterns from `.dotagents/.agentignore` (newline-separated glob patterns, gitignore-like)
- Supports per-provider `[providers.<name>.ignore]` config with `disabled` toggle and `template`/`target` paths
- Creates `ignore.hbs` templates and updated `provider.toml` for 11 providers: opencode, auggie, autohand, cline, cursor, gemini, goose, junie, kilocode, pi, qwen
- Integrates ignore into `init` CLI (`--features agent-ignore`) and TUI wizard ("Ignore Patterns" pre-selected option)
- Scaffolds default `.agentignore` with common patterns (`node_modules/`, `.git/`, `target/`, `.env`)
- Uses single-phase rendering (no frontmatter) and includes ignore file paths in `.gitignore` fence tracking
- Adds `strum`/`strum_macros` for derive macros on `Feature` enum

## Testing

- Unit tests for `IgnoreFeature` roundtrip, pattern parsing, and edge cases (empty/missing `.agentignore`)
- Unit tests for config parsing and merging (`Features::merge` with ignore field)
- Unit tests for init with ignore feature selected/deselected
- E2E tests for ignore feature deploy flow (writes correct provider-specific files, respects `disabled` flag)
- E2E tests for init with `--features agent-ignore` flag
- `mise check` (cargo fmt + clippy) passes
- `mise tests` (unit + integration + e2e) passes

## Release Notes

* **New Features**
  * Added ignore patterns feature to exclude files and directories from AI tools during initialization and deployment
  * Enabled support for generating provider-specific ignore files (`.agentignore`, `.cursorignore`, `.augmentignore`, etc.) across 11+ AI providers

* **Chores**
  * Updated CLI feature naming conventions for consistency and clarity

[Review Change Stack](https://app.coderabbit.ai/change-stack/soorya-u/dotagents/pull/159?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/constants/mocks.rs [hunk 1]
```diff
@@ -70,6 +70,8 @@ pub(crate) const MCP_MOCK: &str = r#"{
 }
 "#;
 
+pub(crate) const AGENTIGNORE_MOCK: &str = ".env.prod";
+
```
[the remark this answers is not in the document]
Comment (soorya-u) [reply src/constants/mocks.rs:74]: this is just an example and i think this is fine for now
